### ISSUE TITLE
Assert::integerish() with floats/numeric will always evaluate to false

### ISSUE THREAD

**Issue by dragosprotung** [phpstan/phpstan-webmozart-assert#32]
Calling `Assert::integerish(1.000)` with a float or numeric the following error is reported:

`Call to static method Webmozart\Assert\Assert::integerish() with float will always evaluate to false.`

```php
Assert::integerish(1.000); // OK
Assert::integerish(1.001); // not OK
```

**Comment by herndlm** [comment]
It has been quite a while since you reported this. I just tried it out and it now throws only errors if phpstan-strict-rules is used too. Then you'll get a "Call to function is_numeric() with xxx will always evaluate to true." https://phpstan.org/r/2c6e2d18-b7b0-4c5d-a0b8-0fa61527e80a
Which makes sense IMO. Do you know if this is still a bug/relevant?

UPDATE: Ah, I think I got it now. This is about using float type variables via integerish, right? But even in that case, it would always evaluate to true for values that can be interpreted as float or ints as in https://www.php.net/manual/en/language.types.numeric-strings.php

UPDATE2: I think it works fine if variables are passed, but with your constant floats it doesn't, yeah. Not sure if this is really still a problem though. I think I found one thing that can be improved and will continue looking

**Comment by dragosprotung** [comment]
I should have probably not do an example with constants. Here is another example:

```php
function test(float $float): void 
{
    // Call to static method Webmozart\Assert\Assert::integerish() with float will always evaluate to true.
    Assert::integerish($float);
}
```

**Comment by herndlm** [comment]
Thx, my assumption was right then. I'll try to look into that soon
